### PR TITLE
Feature/pattern expression ref local variable

### PR DESCRIPTION
**Pull request by Shylock-Hg**
## What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### Description:
Support pattern expression reference local defined variable, implement this by filter path in list(the result of pattern expressioin)

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/graph/util/ParserUtil.cpp [hunk 1]
```diff
@@ -14,21 +14,21 @@ namespace graph {
 void ParserUtil::rewriteLC(QueryContext *qctx,
                            ListComprehensionExpression *lc,
                            const std::string &oldVarName) {
-  const auto &newVarName = qctx->vctx()->anonVarGen()->getVar();
-  qctx->ectx()->setValue(newVarName, Value());
+  qctx->vctx()->anonVarGen()->createVar(oldVarName);
+  qctx->ectx()->setValue(oldVarName, Value());
```
Comment (jievince) [inline src/graph/util/ParserUtil.cpp:18]: Can you explain this?

Comment (Shylock-Hg) [reply src/graph/util/ParserUtil.cpp:18]: In List Comprehesion `[i in [...] | ...]`, the local defined variable `i` will be used in sub-expr, so I keep its origin name.

Comment (jievince) [reply src/graph/util/ParserUtil.cpp:18]: What if the local var name same with the global var?

Comment (Shylock-Hg) [reply src/graph/util/ParserUtil.cpp:18]: Shadow global variable.

**Review by jievince — APPROVED** [review]

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**